backend/src/main/kotlin/org/loculus/backend/service/seqsetcitations/SeqSetCitationsDatabaseService.kt

```diff
@@ -314,8 +312,8 @@ class SeqSetCitationsDatabaseService(
             throw NotFoundException("SeqSet $seqSetId, version $version does not exist")
         }
 
-        val now = Clock.System.now().toLocalDateTime(TimeZone.UTC).toJavaLocalDateTime()
-        val sevenDaysAgo = LocalDateTime.parse(now.minusDays(7).toString())
+        val now = dateProvider.getCurrentInstant()
+        val sevenDaysAgo = now.minus(7, DateTimeUnit.DAY, TimeZone.UTC).toLocalDateTime(TimeZone.UTC)
```

Comment:
Would it make sense to move `TimeZone.UTC` to the dateProvider as well so that it holds all related constants? i.e., call dateProvider.getTimeZone() here?